### PR TITLE
 Allow `--use-application-binary` using app-bundles on ios

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

**Review by mravn-google — APPROVED** [review]
Code LGTM. Could we add tests to the `flutter_tools` test suite covering the two sunny cases and the error scenarios?

**Comment by sigurdm** [comment]
Tests added. The code is updated to improve error messaging.
I added `IOSWorkFlow.getPlistValueFromFile` to enable mocking, and adapted all uses of plist_util/getValueFromFile.
PTAL

**Review by mravn-google — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/application_package.dart [hunk 1]
```diff
@@ -145,29 +146,58 @@ bool _isBundleDirectory(FileSystemEntity entity) =>
...
+      final Directory payloadDir = fs.directory(
+        fs.path.join(tempDir.path, 'Payload'),
+      );
+      if (!payloadDir.existsSync()) {
+        printError(
+            'Invalid prebuilt iOS ipa. Does not contain a "Payload" directory.');
+        return null;
+      }
+      try {
+      bundleDir = payloadDir.listSync()
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply packages/flutter_tools/lib/src/application_package.dart]: Done

File: packages/flutter_tools/test/application_package_test.dart [hunk 2]
```diff
@@ -28,6 +37,118 @@ void main() {
...
+    testUsingContext('Error on non-app-bundle folder', () {
+      fs.directory('regular_folder').createSync();
+      final PrebuiltIOSApp iosApp =
+          new IOSApp.fromPrebuiltApp('regular_folder');
+      expect(iosApp, isNull);
+      final BufferLogger logger = context[Logger];
+      expect(
+          logger.errorText, 'Folder "regular_folder" is not an app bundle.\n');
+    }, overrides: overrides);
+
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply packages/flutter_tools/test/application_package_test.dart]: done

File: packages/flutter_tools/test/application_package_test.dart [hunk 3]
```diff
@@ -28,6 +37,118 @@ void main() {
...
+    testUsingContext('Error on no info.plist', () {
+      fs.directory('bundle.app').createSync();
+      final PrebuiltIOSApp iosApp = new IOSApp.fromPrebuiltApp('bundle.app');
+      expect(iosApp, isNull);
+      final BufferLogger logger = context[Logger];
+      expect(
+        logger.errorText,
+        'Invalid prebuilt iOS app. Info.plist does not contain bundle identifier\n',
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply packages/flutter_tools/test/application_package_test.dart]: Fixed. Now reporting a missing Info.plist

**Pull request merged**